lib/node_modules/@stdlib/math/base/special/modf/src/main.c

```diff
@@ -0,0 +1,132 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/modf.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/number/float64/base/from_words.h"
+#include "stdlib/number/float64/base/to_words.h"
+#include "stdlib/constants/float64/pinf.h"
+#include "stdlib/constants/float64/exponent_bias.h"
+#include "stdlib/constants/float64/high_word_exponent_mask.h"
+#include "stdlib/constants/float64/high_word_significand_mask.h"
+
+// 4294967295 => 0xffffffff => 11111111111111111111111111111111
+static const uint32_t ALL_ONES = 4294967295;
+
+/**
+* Decomposes a double-precision floating-point number into integral and fractional parts, each having the same type and sign as the input value, and assigns results to a provided output array.
+*
+* @param x           input value
+* @param frac        destination to store the normalized fraction
+* @param integral    destination to store the exponent
+*
+* @example
+* double x = 3.141592653589793;
+*
+* double integral;
+* double frac;
+* stdlib_base_modf( x, &integral, &frac );
+*/
+void stdlib_base_modf( const double x, double* integral, double* frac ) {
+	uint32_t high;
+	uint32_t low;
+	int32_t exp;
+	int32_t i;
+	double j;
+
+	// Special cases...
+	if ( x < 1.0 ) {
+		if ( x < 0.0 ) {
+			stdlib_base_modf( -x, integral, frac );
+			*frac *= -1.0;
+			*integral *= -1.0;
```

Comment:
```suggestion
			*integral *= -1.0;
			*frac *= -1.0;
```

This makes no difference but shows incorrect interpretation and creates confusion, better fix it.